### DISCUSSION TITLE
creating-a-source-generator-part-6-saving-source-generator-output-in-source-control/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#58] [Announcements]
# creating-a-source-generator-part-6-saving-source-generator-output-in-source-control/

In this post I show how to emit your source generator output to disk, so you can include it in source control and code reviews.

https://andrewlock.net/creating-a-source-generator-part-6-saving-source-generator-output-in-source-control/

**Comment by rsam70** [comment]
Thx, nice article.

**Comment by Kapusch** [comment]
At the moment, generated files can't be explored through Visual Studio for Mac.
Hopefully, `EmitCompilerGeneratedFiles` solved it! Thank you for sharing.

[a comment by an automated account is not shown]

**Comment by TripleView** [comment]
thx,very good~

**Comment by TripleView** [comment]
This article helped me solve a problem that has been bothering me for a few days

**Comment by jeme** [comment]
I am working on making a incremental source generator which takes a template and transform that into a C# Method for that template, e.g.:

MyTemplate.html / MyTemplate.sql etc ->

HtmlTemplates.MyTemplate( ...args );
SqlTemplates.MyTemplate( ...args );

And that works, but I ran into the issue described here where "error CS0111: Type 'ColoursExtensions' already defines a member called 'IsDefined' with the same parameter types ColoursExtensions_EnumExtensions.g.cs(40,28): error CS0111: Type 'ColoursExtensions' already defines a member called 'TryParse' with the same parameter type"

The problem is that if I add:

 <Compile Remove="$(GeneratedFolder)/**/*.cs" />

I can no longer call my generated methods. And I can't figure out a way around this for now.
Any ideas?

**Reply by jeme** [reply]
The problem is that if I add: (I had copied the Compile Remove part in under that, I guess it must have been removed due to the angle-brackets)

**Comment by dahlmax** [comment]
This is a great series of posts. Very well written and covers everything I need to do. 
Has anyone come across an issue with generators in VSCode? I want to generate output files in the IDE that can be browsed and navigated. Using EmitCompilerGeneratedFiles + GeneratedFolder + Compile Remove does produce the output files and allows me to compile using the in-memory files. However, the output files are excluded from the VSCode workspace. This makes them difficult to navigate (although strangely you can use them for breakpoints). The files are effectively hidden, so don't show in Solution Explorer, and you cannot click through the navigation, which is quite inconvenient. Has anyone else come across this issue?

**Reply by Kapusch** [reply]
Hi @dahlmax , I am using VS Code for few weeks now and I remember having been able to reach some generated source code for RelayCommands, by using Command+Click combination in MacOS. Did you give it a try ?
If no success, I assume you could get some help from the .NET CommunityToolkit, via a new discussion.

**Comment by vmachacek** [comment]
I'm wondering if I can remove from compilation the originally generated files so that I can step thru the ones which are outputted into file system. Currently it seems Rider wont allow me to step thru files which are auto-generated, even I can browse them
